### PR TITLE
Remove shots attr from `expval` and `var` operations

### PR DESCRIPTION
**Pull request by paul0403**
**Context:**
As part of the work to support dynamically shaped measurement primitives, shots is now tied to the device through a SSA value, so it can be dynamic. See #1310 .

Correspondingly, the static shots int literal attribute of the `expval` and `var` operations in mlir should be removed. This is more so the case since their shots values are not actually used in their respective runtime CAPIs.

**Description of the Change:**
Remove shots attr from `expval` and `var` operations in mlir, and frontend tracing and jax primitives.

**Benefits:**
Unification of how measurement operations handle shots.

[sc-78845]

### CODE REVIEW

File: mlir/include/Quantum/IR/QuantumOps.td [hunk 1]
```diff
@@ -876,8 +876,8 @@ def ExpvalOp : Measurement_Op<"expval"> {
     }];
 
     let arguments = (ins
-        ObservableType:$obs,
-        OptionalAttr<I64Attr>:$shots
+        ObservableType:$obs
+        //OptionalAttr<I64Attr>:$shots
```
Comment (paul0403) [inline mlir/include/Quantum/IR/QuantumOps.td:880]: do `var`

File: frontend/catalyst/jax_primitives.py [hunk 2]
```diff
@@ -1430,17 +1430,17 @@ def _counts_lowering(
 # expval measurement
 #
 @expval_p.def_abstract_eval
-def _expval_abstract_eval(obs, shots, shape=None):
+def _expval_abstract_eval(obs, shape=None):
```
Comment (paul0403) [inline frontend/catalyst/jax_primitives.py:1433]: changelog

**Comment by paul0403** [comment]
@rauletorresc was this the "some measurements in c++ doesn't actually use shots" you were talking about a while ago?

**Review by erick-xanadu — APPROVED** [review]

File: frontend/catalyst/from_plxpr.py [hunk 3]
```diff
@@ -370,6 +370,8 @@ def interpret_measurement_eqn(self, eqn: jax.core.JaxprEqn):
             mval = bind_flexible_primitive(
                 counts_p, {"shots": device_shots}, obs, shape=shaped_array.shape
             )
+        elif primitive in (expval_p, var_p):
+            mval = expval_p.bind(obs, shape=shaped_array.shape)
```
Comment (erick-xanadu) [inline frontend/catalyst/from_plxpr.py:374]: Should you also bind to var_p?

**Review by erick-xanadu — CHANGES_REQUESTED** [review]

File: frontend/catalyst/from_plxpr.py (hunk 3 above)
Comment (paul0403) [reply frontend/catalyst/from_plxpr.py]: done

**Review by erick-xanadu — APPROVED** [review]

**Pull request merged**